common/src/main/java/org/apache/rocketmq/common/protocol/header/SendMessageResponseHeader.java

```diff
@@ -37,6 +42,37 @@
     public void checkFields() throws RemotingCommandException {
     }
 
+    @Override
+    public void encode(ByteBuf out) {
+        writeIfNotNull(out, "msgId", msgId);
+        writeIfNotNull(out, "queueId", queueId);
+        writeIfNotNull(out, "queueOffset", queueOffset);
+        writeIfNotNull(out, "transactionId", transactionId);
+    }
+
+    @Override
+    public void decode(HashMap<String, String> fields) throws RemotingCommandException {
+        String str = getAndCheckNotNull(fields, "msgId");
```

Comment:
It would be better to add a test to compare the old decode result and fast decode result in case we add a new field later.

Reply:
I found the tests, please ignore it.